www/src/views/shared/collapsible.js

```diff
@@ -1,83 +1,71 @@
 /** @jsx jsx */
 import { jsx } from "theme-ui"
-import { Component } from "react"
+import { useState } from "react"
 import { FaAngleDown, FaAngleUp } from "react-icons/fa"
 
-class Collapsible extends Component {
-  state = {
-    collapsed: false,
-  }
+import { Flex } from "theme-ui"
 
-  handleClick = () => {
-    this.setState({ collapsed: !this.state.collapsed })
-  }
+export default function Collapsible(props) {
+  const { heading, fixed, children } = props
+  const [collapsed, setCollapsed] = useState(false)
 
-  render() {
-    const { heading, fixed, children } = this.props
-    const { collapsed } = this.state
-
-    return (
-      <div
+  return (
+    <div
+      sx={{
+        borderBottom: t => (collapsed ? 0 : `1px solid ${t.colors.ui.border}`),
+        display: collapsed ? false : `flex`,
+        flex: collapsed ? `0 0 auto` : `1 1 auto`,
+        minHeight: fixed ? `${fixed}px` : `initial`,
+        maxHeight: fixed ? `${fixed}px` : `initial`,
+        flexBasis: 0,
+        overflowY: collapsed ? false : `auto`,
+      }}
+    >
+      <Flex
         sx={{
-          borderBottom: t =>
-            collapsed ? 0 : `1px solid ${t.colors.ui.border}`,
-          display: collapsed ? false : `flex`,
-          flex: collapsed ? `0 0 auto` : `1 1 auto`,
-          minHeight: fixed ? `${fixed}px` : `initial`,
-          maxHeight: fixed ? `${fixed}px` : `initial`,
-          flexBasis: 0,
-          overflowY: collapsed ? false : `auto`,
+          flexDirection: `column`,
+          minHeight: `100%`,
+          width: `100%`,
         }}
       >
-        <div
-          css={{
+        <button
+          sx={{
+            alignItems: `center`,
+            color: `textMuted`,
+            cursor: `pointer`,
             display: `flex`,
-            flexDirection: `column`,
-            minHeight: `100%`,
-            width: `100%`,
+            flexShrink: 0,
+            fontWeight: `body`,
+            fontSize: 1,
+            my: 6,
+            mr: 4,
+            p: 0,
+            letterSpacing: `tracked`,
+            textTransform: `uppercase`,
+            background: `none`,
+            border: `none`,
+            "&:hover": {
+              color: `gatsby`,
+            },
+          }}
+          aria-expanded={!collapsed}
+          onClick={() => setCollapsed(!collapsed)}
```

Comment:
https://reactjs.org/docs/faq-functions.html#arrow-function-in-render

> Note:
>
> Using an arrow function in render creates a new function each time the component renders, which may break optimizations based on strict identity comparison.
>
> Is it OK to use arrow functions in render methods?
> Generally speaking, yes, it is OK, and it is often the easiest way to pass parameters to callback functions.
>
> If you do have performance issues, by all means, optimize!

can this be here an performance issue?

Reply:
I guess it will create performance issue only when the user is repeatedly toggling the collapsible on and off, what do you think about it?

Reply:
```suggestion
          onClick={toggleCollapsible}
```
Should I create a toggleCollapsible function using function declaration and then toggle the state there, what are your thoughts on this?

Reply:
@tesseralis

Reply:
Nah, it's really only an issue if you can actually *perceive* a difference. "premature optimization is the root of all evil" after all:

https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578

Reply:
Also, since this is a function component, moving the function out would do nothing. The only time it's reasonable is if you use `useCallback` *and* you're putting it in a place where function identity is important, like the dependency array of a useEffect hook.